alibi/prototypes/protoselect.py

```diff
@@ -78,59 +78,59 @@ def __init__(self,
         })
 
     def fit(self,  # type: ignore[override]
-            X_ref: Union[list, np.ndarray],
-            Y_ref: Optional[np.ndarray] = None,
-            X: Optional[Union[list, np.ndarray]] = None) -> 'ProtoSelect':
+            X: Union[list, np.ndarray],
+            y: Optional[np.ndarray] = None,
+            Z: Optional[Union[list, np.ndarray]] = None) -> 'ProtoSelect':
         """
-        Fit the summariser by setting the reference dataset. This step forms the kernel matrix in memory
-        which has a shape of `Nx x Ny`, where `Nx` is the number of instances in `X` and `Ny` is the
-        number of instances in `Y`.
+        Fit the summariser. This step forms the kernel matrix in memory which has a shape of `NX x NZ`,
+        where `NX` is the number of instances in `X` and `NZ` is the number of instances in `Z`.
```

Comment:
Add that the kernel matrix will be of size `NX x NX` if `Z` is not provided (can't expect users to read all of the docstring). Even better, I would first mention the usual use case `NX x NX` and then mention the other case where `Z` is not None.